src/smart_controller_ui.cpp

```diff
@@ -0,0 +1,179 @@
+
+
+static catacurses::window init_window()
+{
+    const int width = smart_controller_ui::WIDTH;
+    const int height = smart_controller_ui::HEIGHT;
+    const point p( std::max( 0, ( TERMX - width ) / 2 ), std::max( 0, ( TERMY - height ) / 2 ) );
+    return catacurses::newwin( height, width, p );
+}
+
+smart_controller_settings::smart_controller_settings( bool &enabled, int &battery_lo,
+        int &battery_hi ) : enabled(
+                enabled ), battery_lo( battery_lo ), battery_hi( battery_hi ) {}
+
+smart_controller_ui::smart_controller_ui( smart_controller_settings initial_settings ) :
+    win( init_window() ), input_ctx( "SMART_ENGINE_CONTROLLER" ), settings( initial_settings )
+{
+
+    input_ctx.register_directions();
+    input_ctx.register_action( "QUIT" );
+    input_ctx.register_action( "CONFIRM" );
+    input_ctx.register_action( "NEXT_TAB" );
+}
+
+void smart_controller_ui::refresh()
+{
+    werase( win );
+    draw_border( win );
+
+    const auto gray = c_light_gray;
+    const auto white = c_white;
+    //    const auto green = c_green;
+    const auto lgreen = c_light_green;
+
+    // header
+    const std::string title =  _( "Smart Engine Controller ® Interface" );
+    mvwprintz( win, point( ( WIDTH - title.length() ) / 2, 1 ), white, title );
+    mvwhline( win, point( 1, 2 ), LINE_OXOX, WIDTH - 2 );
+
+    // for menu items, y points to the center of the menu item vertical space
+    int y = 3 + MENU_ITEM_HEIGHT / 2;
+
+    // enabled flag
+    mvwprintz( win, point( LEFT_MARGIN, y ), gray,  "[ ]" );
+    if( settings.enabled ) {
+        mvwprintz( win, point( LEFT_MARGIN + 1, y ), white, "X" );
+    }
+    mvwprintz( win, point( LEFT_MARGIN + 4, y ), selection == 0 ? hilite( white ) : gray,
+               _( "Enabled" ) );
+
+    // battery % slider
+    y += MENU_ITEM_HEIGHT;
+    mvwprintz( win, point( LEFT_MARGIN, y - 1 ), selection == 1 ? white : gray,
+               _( "Electric motor use (%% battery)" ) );
+
+    int lo_slider_x = settings.battery_lo * SLIDER_W / 100;
+    int hi_slider_x = settings.battery_hi * SLIDER_W / 100 ;
+    // print selected % numbers
+    std::string battery_low_text = string_format( "%d%%", settings.battery_lo );
+    mvwprintz( win, point( LEFT_MARGIN + lo_slider_x - battery_low_text.length() + 1, y + 2 ),
+               selection == 1 && slider == 0 ? hilite( white ) : c_red, battery_low_text );
+    mvwprintz( win, point( LEFT_MARGIN + hi_slider_x, y + 2 ),
+               selection == 1 && slider == 1 ? hilite( white ) : lgreen, "%d%%", settings.battery_hi );
+    // draw slider horizontal line
+    for( int i = 0; i < SLIDER_W; ++i ) {
+        //        nc_color col = i < lo_slider_x ? c_light_red : (i < hi_slider_x ? c_yellow : green);
+        nc_color col = selection == 1 ? white : gray;
+        mvwprintz( win, point( LEFT_MARGIN + i, y + 1 ), col, "-" );
+    }
+    // print LO and HI on the slider line
+    std::string lo_text = _( "LO" );
```

Comment:
Please add translator comments to this and to "HI" below.

Reply:
Wow, didn't know about translator comments. Will do!